src/components/property-tree/property-tree.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from "react";
+// @ts-ignore
+import SortableTree from "react-sortable-tree";
+
+import { ValidChildAsset } from "../../utils/test-fixtures";
+
+import { Asset } from "@mtfh/common/lib/api/asset/v1/types";
+import "react-sortable-tree/style.css";
+
+interface PropertyTreeProps {
+  asset: Asset;
+  childAssets: Asset[] | ValidChildAsset[] | undefined;
+}
+
+interface TreeAsset {
+  title: React.ReactNode;
+  children: Array<TreeAsset> | null | { title: string; children: null };
+  expanded?: boolean;
+}
+
+export const PropertyTree = ({ asset, childAssets }: PropertyTreeProps): JSX.Element => {
+  const excludedTreeAssets = "656feda1-896f-b136-da84-163ee4f1be6c"; // Hackney Homes
+
+  const [treeViewData, setTreeViewData] = useState<TreeAsset[]>([]);
+
+  const addChildrenAssets = (): TreeAsset[] => {
+    const childrenAssets: TreeAsset[] = [];
+
+    if (childAssets) {
+      for (const [i, v] of childAssets.entries()) {
+        if (i < 5) {
+          childrenAssets.push(generateNode(v.assetAddress.addressLine1, [], v.id));
+        } else if (i === 5) {
+          childrenAssets.push({ title: "<< MORE.... >>", children: null });
+        } else {
+          break;
+        }
+      }
+    }
+
+    return childrenAssets;
+  };
+
+  const generateNode = (name: string, childList: TreeAsset[], id: string): TreeAsset => {
+    const node = (
+      <a className="lbh-link govuk-link" href={`/property/${id}`}>
+        {name}
+      </a>
+    );
+
+    return { title: node, children: childList, expanded: false };
+  };
+
+  const generatePrinciple = (asset: Asset, childNodes: TreeAsset[]): TreeAsset => {
+    if (asset.assetLocation?.parentAssets?.length) {
+      return {
+        title: <span>Principle</span>,
+        children: [
+          {
+            title: `${asset.assetAddress.addressLine1} (this asset)`,
+            children: childNodes,
+            expanded: true,
+          },
+        ],
+        expanded: true,
+      };
+    }
+
+    return {
+      title: <span>Hackney</span>,
+      children: [
+        {
+          title: `${asset.assetAddress.addressLine1} (this asset)`,
+          children: childNodes,
+          expanded: true,
+        },
+      ],
+      expanded: true,
+    };
+  };
```

Comment:
The code appears to be identical except for the title. You could instead add a condition only for the title.

Reply:
Good spot, addressed